workflow/kfdrc_production_amplicon_architect.cwl

```diff
@@ -0,0 +1,286 @@
+    in:
+      input_reads: tumor_align_file
+      reference: reference
+    out: [bam_file]
+
+  samtools_normal_cram_to_bam:
+    run: ../tools/samtools_calmd.cwl
+    when: $(inputs.input_reads != null && inputs.input_reads.nameext == ".cram")
+    in:
+      input_reads: normal_align_file
+      reference: reference
+    out: [bam_file]
+
+  untar_data_repo:
+    run: ../tools/untar_gzip.cwl
+    in:
+      input_tar: aa_data_repo
+      out_dir_name:
+        valueFrom: ${return "data_repo"}
+    out: [output]
+
+
+  cnvkit_batch:
+    run: ../tools/cnvkit_batch_only.cwl
+    when: $(inputs.cnvkit_cnn != null || inputs.input_control != null)
+    in:
+      input_sample:
+        source: [samtools_tumor_cram_to_bam/bam_file, tumor_align_file]
+        pickValue: first_non_null
+      input_control:
+        source: [samtools_normal_cram_to_bam/bam_file, normal_align_file]
+        pickValue: first_non_null
+      reference:
+        # samtools needs fasta, can't use fasta if cnn file provided
+        source: [cnvkit_cnn, reference]
+        valueFrom: "${ if (self[0] == null){ return self[1]; } else{ return null;\
+          \ } }"
+      cnvkit_cnn: cnvkit_cnn
+      annotation_file: annotation_file
+      male_input_flag: male_input_flag
+      output_basename: output_basename
+    out: [output_cnr, output_cns, output_scatter, output_diagram, output_cnn]
+
+  cns_to_aa_bed:
+    run: ../tools/cns_to_aa_bed.cwl
+    in:
+      input_cns:
+        source: [cnvkit_batch/output_cns, cnvkit_cns]
+        pickValue: first_non_null
+    out: [aa_cns_bed]
+
+  prepare_aa:
+    run: ../tools/prepare_aa.cwl
+    in:
+      data_repo: untar_data_repo/output
+      data_ref_version: aa_data_ref_version
+      sample: output_basename
+      sorted_bam:
+        source: [samtools_tumor_cram_to_bam/bam_file, tumor_align_file]
+        pickValue: first_non_null
+      cnv_bed: cns_to_aa_bed/aa_cns_bed
+    out: [aa_cnv_seeds, coverage_stats]
+
+  amplicon_architect:
+    run: ../tools/amplicon_architect.cwl
+    in:
+      data_repo: untar_data_repo/output
+      data_ref_version: aa_data_ref_version
+      bam:
+        source: [samtools_tumor_cram_to_bam/bam_file, tumor_align_file]
+        pickValue: first_non_null
+      output_basename: output_basename
+      coverage_stats: prepare_aa/coverage_stats
+      aa_bed: prepare_aa/aa_cnv_seeds
+      mosek_license_file: mosek_license_file
+    out: [summary, cycles, graph, sv_pdf, sv_png]
+
+  amplicon_classifier:
+    run: ../tools/aa_classifier.cwl
+    in:
+      data_repo: untar_data_repo/output
+      data_ref_version: aa_data_ref_version
+      cycles: amplicon_architect/cycles
+      graph: amplicon_architect/graph
+    scatter: [cycles, graph]
+    scatterMethod: dotproduct
+    out: [amplicon_classification_profiles, gene_list, ecDNA_cts]
+
+$namespaces:
+  sbg: https://sevenbridges.com
+hints:
+- class: "sbg:maxNumberOfParallelInstances"
+  value: 3
```

Comment:
Looking at your test runs you never use more than 1 instance at a time. Do you really need three parallel for this?

Reply:
I think in one of the tests I do use at least 2 instances - to convert the tumor and normal cram to bam at the same time

Reply:
Sounds good.